lib/view_component/isolated_i18n.rb

```diff
@@ -0,0 +1,44 @@
+require 'set'
+require 'action_view/helpers/translation_helper'
+
+module ViewComponent::IsolatedI18n
+  def self.included(base)
+    base.class_attribute :isolated_i18n_backend, instance_writer: false, instance_predicate: false
+    base.extend ClassMethods
+  end
+
+  Cache = Set.new
+
+  def translate(key = nil, locale: nil, **options) # TODO deprecate :raise
+    locale ||= ::I18n.locale
+    backend = isolated_i18n_backend
+
+    result = catch(:exception) do
+      if key.is_a?(Array)
+        key.map { |k| backend.translate(locale, k, options) }
+      else
+        backend.translate(locale, key, options)
+      end
+    end
+
+    if result.is_a?(I18n::MissingTranslation)
+      handle_exception(false, result, locale, key, options)
```

Comment:
I wonder if we could fall-back to global translations for this case so you don't have to keep track of "global" vs "local" translations? What are your thoughts?

Reply:
We could even go with the [Chain](https://github.com/ruby-i18n/i18n/blob/master/lib/i18n/backend/chain.rb) backend and avoid the throw/catch game altogether, but I thought that there's some value in strictly using the sidecar translations.

No strong feelings though, with regards to this I'm also open to falling back when a missing-translation is thrown, or the chain backend, or even a "switch" parameter like `fallback_to_global: true` etc.